pkgs/applications/misc/curaengine/default.nix

```diff
@@ -1,18 +1,38 @@
-{ lib, stdenv, fetchFromGitHub, cmake, libarcus, stb, protobuf }:
+{ stdenv
+, lib
+, fetchFromGitHub
+, cmake
+
+# Dependencies
+, clipper
+, libarcus
+, protobuf
+, rapidjson
+, stb
+}:
 
 stdenv.mkDerivation rec {
   pname = "curaengine";
-  version = "4.13.1";
+  version = "5.0.0";
 
   src = fetchFromGitHub {
     owner = "Ultimaker";
     repo = "CuraEngine";
-    rev = version;
-    sha256 = "sha256-dx0Q6cuA66lG4nwR7quW5Tvs9sdxjdV4gtpxXirI4nY=";
+    # TODO: Switch back to `rev = version;` on the next release that has this fix:
+    #       https://github.com/Ultimaker/CuraEngine/commit/bb9ad578abc949c474db7a0677a355ad379ae585#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aL262
+    #       for https://github.com/Ultimaker/CuraEngine/issues/1650
```

Comment:
> We're in the process of switching to Conan as our dependency manager.

I guess this means we'll have to build Cura using Conan at some point in the future?

Reply:
Possibly.

Cura's switch to Conan, especially in `curaengine`, requires various patches because Conan seems to generate CMake targets that are not available without Conan.

Yesterday I managed to hack around that for `curaengine`, but it's more patching of upstream `CMakeLists.txt` files than I like.

That said, I don't know how nixpkgs can use Conan in an offline fashion (and am not familiar with it at all yet).

Reply:
> That said, I don't know how nixpkgs can use Conan in an offline fashion (and am not familiar with it at all yet).

Looks like you just can't. :|

There's an open issue for it and it suggests manually coping the cache after running it on another machine which is online, which doesn't fit well.

https://github.com/conan-io/conan/issues/5575#issuecomment-518144576